File: Part 7 - Natural Language Processing/Section 36 - Natural Language Processing/natural_language_processing_MINE.py
# Natural Language Processing

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Restaurant_Reviews.tsv', delimiter = '\t', quoting = 3)

# Cleaning the texts
# Importing library used for cleaning text
import re
import nltk
nltk.download('stopwords')
from nltk.corpus import stopwords
from nltk.stem.porter import PorterStemmer
corpus = []
for i in range(0, 1000):
     # Removing all symbols other then a-z A-Z
     review = re.sub('[^a-zA-Z]', ' ', dataset['Review'][i])
     # Making all symbols lower case
     review = review.lower()
     # Spliting string into list of words
     review = review.split()
     ps = PorterStemmer()
     # Removing all unimportant words from list and stemming
     review = [ps.stem(word) for word in review if not word in set(stopwords.words('english'))]
     # Joining list into string
     review = ' '.join(review)
     corpus.append(review)
     
# Creating the Bag of Words model
from sklearn.feature_extraction.text import CountVectorizer
cv = CountVectorizer(max_features = 1500)
X = cv.fit_transform(corpus).toarray()
y = dataset.iloc[:, 1].values

# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.20, random_state = 0)

# Fitting Naive Bayes to the Training set
from sklearn.naive_bayes import GaussianNB
classifier = GaussianNB()
classifier.fit(X_train, y_train)

# Predicting the Test set results
y_pred = classifier.predict(X_test)

# Making the Confusion Matrix
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test, y_pred)

File: Part 7 - Natural Language Processing/Section 36 - Natural Language Processing/natural_language_processing_MINE.R
# Natural Language Processing

# Importing the dataset
dataset = read.delim('Restaurant_Reviews.tsv', quote = '', stringsAsFactors = FALSE)

# Cleaning the texts 
# install.packages('tm')
# install.packages('SnowballC')
library(tm)
library(SnowballC)
corpus = VCorpus(VectorSource(dataset$Review))
# Lower case
corpus = tm_map(corpus, content_transformer(tolower))
# Remowing numbers
corpus = tm_map(corpus, removeNumbers)
# Removing punctuations
corpus = tm_map(corpus, removePunctuation)
# Removing un relevant words
corpus = tm_map(corpus, removeWords, stopwords())
# Stemming
corpus = tm_map(corpus, stemDocument)
# Removing extra spaces
corpus = tm_map(corpus, stripWhitespace)
